src/Algebra/Morphism/Structures.agda

```diff
@@ -177,6 +177,90 @@ module GroupMorphisms (G₁ : RawGroup a ℓ₁) (G₂ : RawGroup b ℓ₂) wher
     open IsMonoidIsomorphism isMonoidIsomorphism public
       using (isRelIsomorphism)
 
+------------------------------------------------------------------------
+-- Morphisms over near-semiring-like structures
+------------------------------------------------------------------------
+
+module NearSemiringMorphisms (R₁ : RawNearSemiring a ℓ₁) (R₂ : RawNearSemiring b ℓ₂) where
+
+  open RawNearSemiring R₁ renaming
+    ( Carrier to A; _≈_ to _≈₁_
+    ; +-rawMonoid to +-rawMonoid₁
+    ; *-rawMagma to *-rawMagma₁)
+
+  open RawNearSemiring R₂ renaming
+    ( Carrier to B; _≈_ to _≈₂_
+    ; +-rawMonoid to +-rawMonoid₂
+    ; *-rawMagma to *-rawMagma₂)
+
+  private
+    module + = MonoidMorphisms +-rawMonoid₁ +-rawMonoid₂
+    module * = MagmaMorphisms *-rawMagma₁ *-rawMagma₂
+
+  open MorphismDefinitions A B _≈₂_
+  open FunctionDefinitions _≈₁_ _≈₂_
+
+  record IsNearSemiringHomomorphism (⟦_⟧ : A → B) : Set (a ⊔ ℓ₁ ⊔ ℓ₂) where
+    field
+      +-isMonoidHomomorphism : +.IsMonoidHomomorphism  ⟦_⟧
+      *-isMagmaHomomorphism  : *.IsMagmaHomomorphism ⟦_⟧
+
+    open +.IsMonoidHomomorphism +-isMonoidHomomorphism renaming
+      (homo to +-homo; ε-homo to 0#-homo) public
+
+    open *.IsMagmaHomomorphism *-isMagmaHomomorphism renaming
+      (homo to *-homo) public
+
+  record IsNearSemiringMonomorphism (⟦_⟧ : A → B) : Set (a ⊔ ℓ₁ ⊔ ℓ₂) where
+    field
+      isNearSemiringHomomorphism : IsNearSemiringHomomorphism ⟦_⟧
+      injective          : Injective ⟦_⟧
+
+    open IsNearSemiringHomomorphism isNearSemiringHomomorphism public
+
+    +-isMonoidMonomorphism : +.IsMonoidMonomorphism ⟦_⟧
+    +-isMonoidMonomorphism = record
+      { isMonoidHomomorphism = +-isMonoidHomomorphism
+      ; injective           = injective
+      }
+
+    *-isMagmaMonomorphism : *.IsMagmaMonomorphism ⟦_⟧
+    *-isMagmaMonomorphism = record
+      { isMagmaHomomorphism = *-isMagmaHomomorphism
+      ; injective            = injective
+      }
+
+    -- TODO: Why isRelMonomorphism specifically here?
```

Comment:
Because it's the only thing not exported by `open IsNearSemiringHomomorphism isNearSemiringHomomorphism public` above